File: chap4/NombreDea.c
// Ces programmes sont sous licence CeCILL-B V1.
#include <stdio.h>
#include <string.h>

static int nombreDea (const char s[])
{
   int i, n;
   n = 0;
   for (i = 0; i <= (int)strlen(s) - 1;i = i + 1) 
   {
      if (s[i] == 'a') 
      {
         n = n + 1;
      }
   }
   return n;
}

int main() 
{
   printf("%d\n",nombreDea("abracadabra"));
}

File: chap4/NombreDea.java
// Ces programmes sont sous licence CeCILL-B V1.
class NombreDea {

  static int nombreDea (String s) {
    int i, n;
    n = 0;
    for (i = 0; i <= Isn.stringLength(s) - 1;i = i + 1) {
      if (Isn.stringEqual(Isn.stringNth(s,i), "a")) {
        n = n + 1;}}
    return n;}

  public static void main (String [] args) {
    System.out.println(nombreDea("abracadabra"));}}
